File: rules-cli/src/main/java/PrettyPrinter.java
import com.github.iamrenny.ruleflow.RuleFlowLanguageBaseListener;
import com.github.iamrenny.ruleflow.RuleFlowLanguageParser;
import org.antlr.v4.runtime.TokenStream;

public class PrettyPrinter extends RuleFlowLanguageBaseListener {
    private final TokenStream tokens;
    private int indentLevel = 0;

    public PrettyPrinter(TokenStream tokens) {
        this.tokens = tokens;
    }

    @Override
    public void enterWorkflow(RuleFlowLanguageParser.WorkflowContext ctx) {
        System.out.print("workflow ");
        indentLevel++;
    }

    @Override
    public void exitWorkflow(RuleFlowLanguageParser.WorkflowContext ctx) {
        indentLevel--;
    }

    @Override
    public void enterWorkflow_name(RuleFlowLanguageParser.Workflow_nameContext ctx) {
        System.out.println(ctx.getText());
    }

    @Override
    public void enterRulesets(RuleFlowLanguageParser.RulesetsContext ctx) {
        printIndent();
        System.out.println("ruleset " + ctx.name().getText());
        indentLevel++;
    }

    @Override
    public void exitRulesets(RuleFlowLanguageParser.RulesetsContext ctx) {
        indentLevel--;
    }

    @Override
    public void enterRules(RuleFlowLanguageParser.RulesContext ctx) {
        printIndent();
        String exprTokens = tokens.getText(ctx.getStart(), ctx.getStop());
        System.out.println(exprTokens);
        indentLevel++;
    }

    public void exitRules(RuleFlowLanguageParser.RulesContext ctx) {
        indentLevel--;
    }

    public void enterDefault_result(RuleFlowLanguageParser.Default_clauseContext ctx) {
        printIndent();
        System.out.println("default " + ctx.getText());
    }

    public void exitDefault_result(RuleFlowLanguageParser.Default_clauseContext ctx) {
        indentLevel--;
        printIndent();
        System.out.println("end");
    }

    protected void printIndent() {
        for (int i = 0; i < indentLevel; i++) {
            System.out.print("  ");
        }
    }
}

File: rules-cli/src/main/kotlin/PrettyPrinter.kt
import com.github.iamrenny.ruleflow.RuleFlowLanguageBaseListener
import com.github.iamrenny.ruleflow.RuleFlowLanguageParser
import org.antlr.v4.runtime.TokenStream

open class PrettyPrinter(val tokens: TokenStream) : RuleFlowLanguageBaseListener() {
    var indentLevel: Int = 0

    override fun enterWorkflow(ctx: RuleFlowLanguageParser.WorkflowContext) {
        print("workflow ")
        indentLevel++
    }

    override fun exitWorkflow(ctx: RuleFlowLanguageParser.WorkflowContext) {
        indentLevel--
    }

    override fun enterWorkflow_name(ctx: RuleFlowLanguageParser.Workflow_nameContext) {
        println(ctx.text)
    }

    override fun enterRulesets(ctx: RuleFlowLanguageParser.RulesetsContext) {
        printIndent()
        println("ruleset ${ctx.name().text}")
        indentLevel++
    }

    override fun exitRulesets(ctx: RuleFlowLanguageParser.RulesetsContext) {
        indentLevel--
    }

    override fun enterRules(ctx: RuleFlowLanguageParser.RulesContext) {
        printIndent()
        val exprTokens = tokens.getText(ctx.start, ctx.stop)
        println(exprTokens)
        indentLevel++
    }

    override fun exitRules(ctx: RuleFlowLanguageParser.RulesContext) {
        indentLevel--
    }


    override fun enterDefault_result(ctx: RuleFlowLanguageParser.Default_resultContext) {
        printIndent()
        println("default ${ctx.text}")
    }

    override fun exitDefault_result(ctx: RuleFlowLanguageParser.Default_resultContext?) {
        indentLevel--
        printIndent()
        println("end")
    }
    open fun printIndent() {
        repeat(indentLevel) { print("  ") }
    }
}